src/csg_tree.cpp

```diff
@@ -206,15 +206,15 @@ std::shared_ptr<CsgLeafNode> CsgLeafNode::Compose(
   Manifold::Impl combined;
   combined.epsilon_ = epsilon;
   combined.tolerance_ = tolerance;
-  combined.vertPos_.resize(numVert);
-  combined.halfedge_.resize(2 * numEdge);
-  combined.faceNormal_.resize(numTri);
-  combined.halfedgeTangent_.resize(2 * numEdge);
-  combined.meshRelation_.triRef.resize(numTri);
+  combined.vertPos_.resize_nofill(numVert);
+  combined.halfedge_.resize_nofill(2 * numEdge);
+  combined.faceNormal_.resize_nofill(numTri);
+  combined.halfedgeTangent_.resize_nofill(2 * numEdge);
+  combined.meshRelation_.triRef.resize_nofill(numTri);
   if (numPropOut > 0) {
     combined.meshRelation_.numProp = numPropOut;
     combined.meshRelation_.properties.resize(numPropOut * numPropVert, 0);
-    combined.meshRelation_.triProperties.resize(numTri);
+    combined.meshRelation_.triProperties.resize(numTri, ivec3(0));
```

Comment:
This was causing the test failure. I forgot what `triProperties` is used for and why are we adding it instead of copying them in with a per input mesh offset.

Also, the weird thing is that the bug is order dependent. I am rerunning the tests with shuffling enabled to try and see if there is something else that is similar.

Reply:
Ah, I was mistaken, this is not the issue. Halfedge tangent is the issue. I guess this may be caused by having empty halfedge tangents?